kotlinlib/src/mill/kotlinlib/js/KotlinJSModule.scala

```diff
@@ -0,0 +1,509 @@
+   */
+  protected override def kotlinCompileTask(
+      extraKotlinArgs: Seq[String] = Seq.empty[String]
+  ): Task[CompilationResult] = Task.Anon {
+    kotlinJsCompile(
+      outputMode = OutputMode.KlibDir,
+      allKotlinSourceFiles = allKotlinSourceFiles(),
+      irClasspath = None,
+      librariesClasspath = compileClasspath(),
+      callMain = callMain(),
+      moduleKind = moduleKind(),
+      produceSourceMaps = kotlinJSSourceMap(),
+      sourceMapEmbedSourcesKind = kotlinJSSourceMapEmbedSources(),
+      sourceMapNamesPolicy = kotlinJSSourceMapNamesPolicy(),
+      splitPerModule = splitPerModule(),
+      esTarget = kotlinJSESTarget(),
+      kotlinVersion = kotlinVersion(),
+      destinationRoot = T.dest,
+      extraKotlinArgs = kotlincOptions() ++ extraKotlinArgs,
+      worker = kotlinWorkerTask()
+    )
+  }
+
+  /**
+   * Creates final executable.
+   */
+  def linkBinary: T[CompilationResult] = Task {
+    kotlinJsCompile(
+      outputMode = binaryKindToOutputMode(kotlinJSBinaryKind()),
+      irClasspath = Some(compile().classes),
+      allKotlinSourceFiles = Seq.empty,
+      librariesClasspath = compileClasspath(),
+      callMain = callMain(),
+      moduleKind = moduleKind(),
+      produceSourceMaps = kotlinJSSourceMap(),
+      sourceMapEmbedSourcesKind = kotlinJSSourceMapEmbedSources(),
+      sourceMapNamesPolicy = kotlinJSSourceMapNamesPolicy(),
+      splitPerModule = splitPerModule(),
+      esTarget = kotlinJSESTarget(),
+      kotlinVersion = kotlinVersion(),
+      destinationRoot = T.dest,
+      extraKotlinArgs = kotlincOptions(),
+      worker = kotlinWorkerTask()
+    )
+  }
+
+  // endregion
+
+  // region private
+
+  private def createKlib(destFolder: os.Path, irPathRef: PathRef): PathRef = {
+    val outputPath = destFolder / s"${moduleName()}.klib"
+    Jvm.createJar(
+      outputPath,
+      Agg(irPathRef.path),
+      mill.api.JarManifest.MillDefault,
+      fileFilter = (_, _) => true
+    )
+    PathRef(outputPath)
+  }
+
+  private[kotlinlib] def kotlinJsCompile(
+      outputMode: OutputMode,
+      allKotlinSourceFiles: Seq[PathRef],
+      irClasspath: Option[PathRef],
+      librariesClasspath: Agg[PathRef],
+      callMain: Boolean,
+      moduleKind: ModuleKind,
+      produceSourceMaps: Boolean,
+      sourceMapEmbedSourcesKind: SourceMapEmbedSourcesKind,
+      sourceMapNamesPolicy: SourceMapNamesPolicy,
+      splitPerModule: Boolean,
+      esTarget: Option[String],
+      kotlinVersion: String,
+      destinationRoot: os.Path,
+      extraKotlinArgs: Seq[String],
+      worker: KotlinWorker
+  )(implicit ctx: mill.api.Ctx): Result[CompilationResult] = {
+    val kotlinVersionSegments = kotlinVersion.split("\\.")
+    if (
+      kotlinVersionSegments(0).toInt == 2 || (
+        kotlinVersionSegments(0).toInt == 1
+          && (kotlinVersionSegments(1).toInt <= 7
+            || (kotlinVersionSegments(1).toInt == 8 && kotlinVersionSegments(2).toInt < 20))
+      )
+    ) {
```

Comment:
Could this be written better as a pattern match?

```scala
val versionAllowed = kotlinVersion.split("\\.").map(_.toInt) match{
  case Array(1, 8, z) => z >= 20
  case Array(1, y, _) => y >= 0
  case Array(2, _, _) => false
}
if (!versionAllowed) return Result.Aborted
```

Reply:
Thanks, updated.

I never used Scala, so my usage of the standard library/language itself is probably far from idiomatic, especially on the pattern matching side.